src/mo_frontend/effect.ml

```diff
@@ -92,7 +111,9 @@ let rec infer_effect_exp (exp:Syntax.exp) : T.eff =
     let e1 = effect_exp exp1 in
     let e2 = effect_cases cases in
     max_eff e1 e2
-  | AsyncE _ ->
+  | AsyncE (T.Fut, _, _) ->
+    T.Await
```

Comment:
Here too this would ideally be `T.Throw`, right?

Reply:
Yes.